### ISSUE TITLE
Use latest of multiple daily export files

### ISSUE THREAD

**Issue by PalminX** [thoughtgap/arc-data-server#5]
Arc creates updates to daily export files by saving a new json.gz with an appended counter, i.e. there might be several files like this:

2019-03-03.json.gz
2019-03-03 2.json.gz
2019-03-03 3.json.gz
2019-03-03 4.json.gz
2019-03-03 5.json.gz
2019-03-03 6.json.gz

arc-data-server currently just reads the first file, because the filename pattern is yyyy-mm-dd.json

I a case like this, it should only take into account the latest (i.e.  2019-03-03 6.json.gz), because it contains the most recent data.

(might need to check with Matt if the current behaviour is by design or if this is going to change, because these loads of duplicates really clutter up the daily export directory)

**Comment by thoughtgap** [comment]
For me, the files in yyyy-mm-dd.json format are always the latest daily exports:

![image](https://user-images.githubusercontent.com/3226639/53978181-e3de8400-410a-11e9-87ce-8d1150a7616b.png)

Is this different for you?

If yes, I would consider building this as a feature into #1

**Comment by PalminX** [comment]
This was how it looked for me:

    2019-03-03 2.json.gz    318.768    2019-03-05 17:08    -a--
    2019-03-03 3.json.gz    318.760    2019-03-05 20:38    -a--
    2019-03-03 4.json.gz    318.778    2019-03-05 22:57    -a--
    2019-03-03 5.json.gz    318.772    2019-03-06 04:38    -a--
    2019-03-03 6.json.gz    318.764    2019-03-06 09:46    -a--
    2019-03-03.json.gz    318.765    2019-03-04 22:51    -a--

I have now deleted these files, and will see if/how they reappear.

Nonetheless, it seems that this could either be handled when handling duplicates, or arc-data-server should only read the most recent file for a certain day.

I had a look at some of the differences, and the thing is that there are slight differences within timeline items, e.g. coordinates which are slightly updated. I would imagine that it is almost impossible to clean up these not-quite-duplicates.

**Comment by PalminX** [comment]
New duplicates have arrived now in Arcs iCloud folder on Windows, e.g.

c:\Users\[...]\iCloudDrive\iCloud~com~bigpaua~LearnerCoacher\Export\JSON\

    [...]
    2019-03-01.json.gz	321.621	2019-03-08 09:41	-a--
    2019-03-01.json(1).gz	321.627	2019-03-09 02:57	-a--
    2019-03-02 2.json.gz	331.852	2019-03-09 02:16	-a--
    2019-03-02.json.gz	331.855	2019-03-08 17:52	-a--

Strangely enough, there seem to be two different updated files, one with a number appended to the date (`2019-03-02 2`), and one with a number appended to the json filename (`2019-03-01.json(1)`).

**Comment by PalminX** [comment]
The updates keep accumulating:

    2019-03-01.json 2.gz	321.630	2019-03-11 19:49	-a--
    2019-03-01 2.json.gz	321.627	2019-03-10 17:50	-a--
    2019-03-01 3.json.gz	321.627	2019-03-09 02:57	-a--
    2019-03-01.json.gz	321.621	2019-03-08 09:41	-a--
    2019-03-02 3.json.gz	331.850	2019-03-11 20:00	-a--
    2019-03-02 2.json(1).gz	331.854	2019-03-10 17:50	-a--
    2019-03-02 2.json.gz	331.852	2019-03-09 02:16	-a--
    2019-03-02.json.gz	331.855	2019-03-08 17:52	-a--

And now there is even a new variant: ```2019-03-01.json 2.gz```

**Comment by PalminX** [comment]
I started working on the Layer1 extraction, and this includes handing of duplicate json.gz files.

See here for my working branch (this is not yet ready for a pull request): https://github.com/PalminX/arc-data-server/tree/Layer1ExtractionWIP

**Comment by thoughtgap** [comment]
This is implemented now as part of #1. Can you please test? You can uncomment [this line](https://github.com/thoughtgap/arc-data-server/blob/104a47cd620bcb48e2af2d8b83c3c779345b9b1c/src/arcFiles.ts#L216) so see how it groups and sorts out which files.

```javascript
//console.log("youngestFileFromList", fileList, youngestFile);
```

**Comment by PalminX** [comment]
Works as expected in the latest version, closing
